### PR TITLE
batches: implement frontend for the webhook banner

### LINKED ISSUE #24310: Webhook not setup banner
### User problem
Nearly every new customer contacts support, reporting that their changesets are not syncing. In the vast majority of cases, their site admin has not setup webhooks. Even some older customers still have not setup webhooks, resulting in [degraded performance](https://docs.sourcegraph.com/batch_changes/references/requirements#batch-changes-effect-on-code-host-rate-limits), and changesets taking up to 8 hours to update versus updating nearly immediately with webhooks.

Users report not knowing about webhooks, or forgetting to set them up. Once instructed to do so, customers successfully setup notebooks. The webhook setup flow is not great, and we have a proposal to fix it, but the primary root cause is that site admins _don't know_ about webhooks.

### Proposed change

[Figma](https://www.figma.com/file/w8O8L0I0k2I5J2qHtqBz0d/webhooks-warning?node-id=0%3A1) | Draft docs page https://github.com/sourcegraph/sourcegraph/pull/25558

If webhooks are not setup for any given codehost connected to the instance:
- non site admin users see a banner on batch changes pages, informing them that changesets are likely not up to date because webhooks are not setup. This banner points to the webhooks docs page, which explains to users they need to ask a site-admin to setup webhooks.
- site admin get a banner, with a link to the docs, and a list of codehosts to configure linking to the codehost config page
- site admins can explicitly opt-out of the webhooks warning for a codehost (in case some organisations have policies against webhooks), by setting a site admin flag. In that case, the banner doesn't show anymore for any user.

- regardless of site admin settings, users see additional [hints](https://www.figma.com/file/w8O8L0I0k2I5J2qHtqBz0d/webhooks-warning?node-id=32%3A2204) if webhooks are not setup

### PR DESCRIPTION
**Pull request by LawnGnome**
This adds the new webhook warning banner based on the backend added in #26759.

Closes #24310.

### CODE REVIEW

File: client/web/src/enterprise/batches/detail/WebhookAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                    {isSiteAdmin ? (
+                        <>
+                            Learn how to{' '}
+                            <Link to="https://docs.sourcegraph.com/batch_changes/how-tos/site_admin_configuration">
+                                configure webhooks
+                            </Link>{' '}
+                            or disable this warning.
+                        </>
+                    ) : (
+                        'Ask your site admin to configure webhooks.'
```
Comment (eseliger) [inline client/web/src/enterprise/batches/detail/WebhookAlert.tsx:75]: should "configure webhooks" also link to the docs so I as a user can link my admin right to it?

File: client/web/src/enterprise/batches/detail/WebhookAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    )}
+                </p>
+                {open && (
+                    <ul>
+                        {nodes.map(codeHost => (
+                            <li key={codeHost.externalServiceKind + codeHost.externalServiceURL}>
+                                <CodeHost {...codeHost} />
+                            </li>
+                        ))}
+                        {hasNextPage && <li key="and-more">and {totalCount - nodes.length} more</li>}
```
Comment (eseliger) [inline client/web/src/enterprise/batches/detail/WebhookAlert.tsx:85]: Nice idea!

**Review by eseliger — APPROVED** [review]

File: client/web/src/enterprise/batches/detail/backend.ts [hunk 3]
```diff
@@ -144,6 +144,16 @@ const batchChangeFragment = gql`
                 createdAt
                 applyURL
             }
+            codeHostsWithoutWebhooks: viewerBatchChangesCodeHosts(first: 5, onlyWithoutWebhooks: true) {
```
Comment (courier-new) [inline client/web/src/enterprise/batches/detail/backend.ts:147]: Not that it super matters, but is there any reason why we request the first 5 but only display the first 3?

File: schema/site.schema.json [hunk 4]
```diff
@@ -381,6 +381,12 @@
         }
       }
     },
+    "batchChanges.disableWebhooksWarning": {
+      "description": "Hides Batch Changes warnings about webhooks not being configured.",
+      "type": "boolean",
+      "group": "BatchChanges",
+      "default": false
+    },
```
Comment (courier-new) [inline schema/site.schema.json:389]: Have we already included in the webhooks docs somewhere a note about using this setting to disable the warning?

**Review by courier-new — APPROVED** [review]
Looks great! Just had two minor questions.

File: client/web/src/enterprise/batches/detail/WebhookAlert.tsx (hunk 1 above)
Comment (LawnGnome) [reply client/web/src/enterprise/batches/detail/WebhookAlert.tsx:75]: Sure, I guess it's possible some developers aren't scared of their admins. :laughing: 

I _really_ wanted to word this "ask your site admin if webhooks are right for you", but it just felt a little too on the nose.

File: client/web/src/enterprise/batches/detail/backend.ts (hunk 3 above)
Comment (LawnGnome) [reply client/web/src/enterprise/batches/detail/backend.ts:147]: Well, you see, I, uh...

...fucked up the constants. It was meant to be 3 in both places. Thanks!

File: schema/site.schema.json (hunk 4 above)
Comment (LawnGnome) [reply schema/site.schema.json:389]: We haven't, but I should do that in this PR. I'll add that!

**Pull request merged**